src/profile-logic/profile-data.js

```diff
@@ -150,6 +150,167 @@ export function getSampleCallNodes(
   });
 }
 
+// This function assumes that the path isn't empty
+export function getFuncIndex(path: CallNodePath): IndexIntoFuncTable {
+  return path[path.length - 1];
+}
+
+// Returns the JS information for a specific stack
+export function getJsInformationForStack(
+  stackIndex: IndexIntoStackTable,
+  { stackTable, frameTable, stringTable }: Thread
+): string {
+  const frameIndex = stackTable.frame[stackIndex];
+  const jsEngineInformationStrIndex = frameTable.implementation[frameIndex];
+  const jsInformation =
+    jsEngineInformationStrIndex === null
+      ? 'interpreter'
+      : stringTable.getString(jsEngineInformationStrIndex);
+
+  return jsInformation;
+}
+
+type TimingsForSpecificPath = {|
+  selfTime: Milliseconds, // time spent in this path excluding children
+  totalTime: Milliseconds, // time spent in this path including children
+  // information about the JavaScript engine used for this path
+  jsEngineInformations: { [key: string]: Milliseconds } | null,
+|};
+
+export type TimingsForPath = {|
+  forPath: TimingsForSpecificPath,
+  forFunc: {|
+    selfTime: Milliseconds, // time spent in this function accross the tree, excluding children
+    totalTime: Milliseconds, // time spent in this function accross the tree, including children
+  |},
```

Comment:
In the future I'd like to compute the `jsEngineInformations` for `forFunc` as well. Not _completely_ sure this is useful though.